modules/explorer-core/src/main/scala/org/ergoplatform/explorer/protocol/dex.scala

```diff
@@ -2,46 +2,70 @@ package org.ergoplatform.explorer.protocol
 import cats.syntax.flatMap._
-import eu.timepit.refined.refineMV
-import eu.timepit.refined.string.HexStringSpec
+import cats.instances.try_._
+import org.ergoplatform.contracts.AssetsAtomicExchangeCompilation
 import org.ergoplatform.explorer.Err.RequestProcessingErr.DexErr.{
   DexBuyOrderAttributesFailed,
   DexSellOrderAttributesFailed
 }
 import org.ergoplatform.explorer.Err.RequestProcessingErr.ContractParsingErr
-import org.ergoplatform.explorer.protocol.utils.{bytesToErgoTree, hexStringBase16ToBytes}
+import org.ergoplatform.explorer.protocol.utils.{
+  bytesToErgoTree,
+  ergoTreeTemplateBytes,
+  hexStringBase16ToBytes
+}
 import org.ergoplatform.explorer.{HexString, TokenId}
 import scorex.util.encode.Base16
 import sigmastate.Values.{ByteArrayConstant, ErgoTree}
+import sigmastate.basics.DLogProtocol.ProveDlog
+import sigmastate.eval.Extensions._
 import sigmastate.{SLong, Values}
 import tofu.Raise.ContravariantRaise
 import tofu.syntax.raise._
 
+import scala.util.Try
+
 object dex {
 
-  // assuming buyer contract from http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L33-L45
   private val SellContractTokenPriceIndexInConstants = 5
+  private val BuyContractTokenIdIndexInConstants     = 6
+  private val BuyContractTokenAmountIndexInConstants = 8
 
-  /** template for the buyer contract as of http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L33-L45
-    */
-  val sellContractTemplate: HexString = HexString(
-    refineMV[HexStringSpec](
-      "eb027300d1eded91b1a57301e6c6b2a5730200040ed801d60193e4c6b2a5730300040ec5a7eded92c1b2a57304007305720193c2b2a5730600d07307"
-    )
-  )
+  def sellContractInstance(tokensPrice: Long): ErgoTree = {
+    import org.ergoplatform.sigma.verified.VerifiedTypeConverters._
+    // since we're only using compiled contracts for constant(parameters) extraction PK value does not matter
+    val anyPk = sigmastate.eval.SigmaDsl.SigmaProp(ProveDlog(constants.group.generator))
+    AssetsAtomicExchangeCompilation.sellerContractInstance(tokensPrice, anyPk).ergoTree
+  }
 
-  // assuming seller contract from http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L33-L45
-  val BuyContractTokenIdIndexInConstants     = 6
-  val BuyContractTokenAmountIndexInConstants = 8
+  val sellContractTemplate: HexString = {
+    // parameter values does not matter, we're extracting ErgoTree template (with placeholders in places of values)
+    val contractErgoTree = sellContractInstance(0L)
+    HexString.fromString[Try](Base16.encode(ergoTreeTemplateBytes(contractErgoTree))).get
```

Comment:
Please avoid unsafe coercion.

Reply:
I suppose you mean adding `RefinementFailed` like in [`HexString.fromString`](http://github.com/ergoplatform/explorer-backend/blob/637d268cc5373209333362169a13754de36c3939/modules/explorer-core/src/main/scala/org/ergoplatform/explorer/package.scala#L187-L194) and handling it at call sites?

I thought of it initially but dismissed the idea. The only reason I see it can fail is if `sellContractInstance()` returns empty/invalid ErgoTree. In this case, it'll blow up on instantiation (in tests, etc. since it's `val`). Which is exactly what we want, in case you update `verified-contract` version. Without changing `verified-contract` version, it unlikely to happen, since returned contract ErgoTree is compiled with macros from Scala source.

Reply:
@greenhat , It's about code discipline -- `.get` makes it dirty. Later someone can declare this val as lazy and we get unexpected behavior somewhere in runtime.

Reply:
Done.